CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# Contributing to Leela Zero
+
+## C++ Usage
+
+Leela Zero is written in C++14, and generally encourages writing in modern C++ style.
+
+This means that:
+
+* The code overwhelmingly uses Almost Always Auto style, and so should you.
+* Prefer range based for and non-member begin/end.
+* You can rely on boost 1.58.0 or later being present.
+* Manipulation of raw pointers is to be avoided as much as possible.
+* Prefer constexpr over defines or constants.
+* Prefer using over typedefs.
+* Prefer uniform initialization.
+* Prefer default initializers for member variables.
+* Aim for const-correctness. Prefer passing non-trivial parameters by const reference.
+* Use header include guards, not #pragma once (pragma once is non-standard, has issues with detecting identical files, and is slower https://gcc.gnu.org/bugzilla/show_bug.cgi?id=58770)
+* config.h is always the first file included.
+* Feel free to use templates, but remember than debugging obscure template metaprogramming bugs is not something people enjoy doing in their spare time.
+* Using exceptions is allowed.
+
+## Code Style
+
+* Look at the surrounding code and the rest of the project!
+* Indentation is 4 spaces. No tabs.
+* public/private/protected access modifiers are de-indented
+* Maximum line length is 80 characters. There are rare exceptions in the code, usually involving user-visible text strings.
+* Ifs are always braced, with very rare exceptions when everything fits on one line and doing it properly makes the code less readable.
+* The code generally avoids any pointer passing and allows non-const references for parameters. Still, for new code it should be preferred to a) put input parameters first b) use return values over output parameters.
```

Comment:
What is the policy on returning multiple values by `pair`s or `tuple`s?

Reply:
I think the code already does it in places. But without C++17 support you may need std::tie instead of being able to use structured bindings.